--- a/plugins/remote-dev-server/bin/launcher.sh
+++ b/plugins/remote-dev-server/bin/launcher.sh
@@ -64,10 +64,6 @@ clean_up_temp()
     rm -f "$TEMP_REMOTE_DEV_PROPERTIES_PATH"
   fi
 
-  if [ -n "${TEMP_JBR:-}" ]; then
-    rm -rf "$TEMP_JBR"
-  fi
-
   if [ -n "${XVFB_PID:-}" ]; then
     kill -9 "$XVFB_PID" >/dev/null 2>&1
   fi
@@ -250,43 +246,6 @@ echo "IDE system directory: $IJ_HOST_SYSTEM_DIR"
 mkdir -p "$IJ_HOST_CONFIG_DIR" || (echo "Failed to create $IJ_HOST_CONFIG_DIR" 1>&2; exit 1)
 mkdir -p "$IJ_HOST_SYSTEM_DIR" || (echo "Failed to create $IJ_HOST_SYSTEM_DIR" 1>&2; exit 1)
 
-# -------------------------------------------------------------------------------------
-# Patch JBR to make self-contained JVM (requires nothing from host system except glibc)
-# -------------------------------------------------------------------------------------
-if [ $IS_DARWIN -ne 1 ]; then
-  SELFCONTAINED_LIBS="$REMOTE_DEV_SERVER_DIR/selfcontained/lib"
-  if [ ! -d "$SELFCONTAINED_LIBS" ]; then
-    echo "ERROR! Unable to locate libraries for self-contained idea distribution. Directory not found: '$SELFCONTAINED_LIBS'." 1>&2
-    exit 1
-  fi
-
-  TEMP_JBR="$IJ_HOST_SYSTEM_DIR/pid.$$.temp.jbr"
-  rm -rf "$TEMP_JBR"
-  cp -r --symbolic-link "$IDE_HOME/jbr" "$TEMP_JBR"
-
-  export "${IDE_PRODUCT_UC}_JDK=$TEMP_JBR"
-
-  patch_bin_file() {
-    file="$1"
-    extra_arg=""
-    if [ "$(basename "$file")" = "java" ]; then
-      extra_arg="\"-Djava.home=$TEMP_JBR\""
-    fi
-    mv "$file" "$file.bin"
-    cat >"$file" <<EOT
-#!/bin/sh
-exec /lib64/ld-linux-x86-64.so.2 --library-path "$SELFCONTAINED_LIBS" "${file}.bin" $extra_arg "\$@"
-EOT
-    chmod 755 "$file"
-  }
-
-  # Hardcoded list copied from LinuxDistributionBuilder.groovy (generateExecutableFilesPatterns)
-  find -L "$TEMP_JBR/bin" -type f -executable | while IFS= read -r line; do patch_bin_file "$line"; done
-  patch_bin_file "$TEMP_JBR/lib/chrome-sandbox"
-  patch_bin_file "$TEMP_JBR/lib/jcef_helper"
-  patch_bin_file "$TEMP_JBR/lib/jexec"
-  patch_bin_file "$TEMP_JBR/lib/jspawnhelper"
-fi
 # -----------------------------
 # Display project trust warning
 # -----------------------------
@@ -366,12 +325,6 @@ fi
 # Replace only default Xmx since users may modify vmoptions file to e.g. make -Xmx even bigger
 sed "s/$IDE_DEFAULT_XMX/-Xmx2048m/g" "$TEMP_VM_OPTIONS" > "${TEMP_VM_OPTIONS}.tmp" && mv "${TEMP_VM_OPTIONS}.tmp" "${TEMP_VM_OPTIONS}"
 
-if [ $IS_DARWIN -ne 1 ]; then
-  # Since TEMP_JBR is built on symlinks, java tries to resolve it and calculates java.home incorrectly
-  # Make sure java.home points to our patched JBR
-  echo "-Djava.home=$TEMP_JBR" >>"$TEMP_VM_OPTIONS"
-fi
-
 export "${IDE_PRODUCT_UC}_VM_OPTIONS=${TEMP_VM_OPTIONS}"
 
 if [ "${REMOTE_DEV_SERVER_JCEF_ENABLED:-}" = "1" ] || [ "${REMOTE_DEV_SERVER_JCEF_ENABLED:-}" = "true" ]; then
